packages/cli/test/unit/decorators/registerController.test.ts

```diff
@@ -4,36 +4,112 @@ jest.mock('@/constants', () => ({
 
 import express from 'express';
 import { agent as testAgent } from 'supertest';
+import { mock } from 'jest-mock-extended';
 
-import { Get, RestController, registerController } from '@/decorators';
-import { AuthService } from '@/auth/auth.service';
-import { mockInstance } from '../../shared/mocking';
+import { ControllerRegistry, Get, Licensed, RestController } from '@/decorators';
+import type { AuthService } from '@/auth/auth.service';
+import type { License } from '@/License';
+import type { SuperAgentTest } from '@test-integration/types';
 
 describe('registerController', () => {
-	@RestController('/test')
-	class TestController {
-		@Get('/unlimited', { skipAuth: true })
-		@Get('/rate-limited', { skipAuth: true, rateLimit: {} })
-		endpoint() {
-			return { ok: true };
+	const license = mock<License>();
+	const authService = mock<AuthService>();
+	let agent: SuperAgentTest;
+
+	beforeEach(() => {
+		jest.resetAllMocks();
+		const app = express();
+		new ControllerRegistry(license, authService).activate(app);
+		agent = testAgent(app);
+	});
+
+	describe('Rate limiting', () => {
+		@RestController('/test')
+		// @ts-expect-error tsc complains about unused class
+		class TestController {
+			@Get('/unlimited')
+			unlimited() {
+				return { ok: true };
+			}
+
+			@Get('/rate-limited', { rateLimit: {} })
```

Comment:
Nit - Not a fan of passing an empty object to mean "enabled with implicit defaults". Maybe this should accept `true | { windowMs: number; limit: number }`